pkg/v1/tkg/clusterclient/clusterclient.go

```diff
@@ -542,12 +542,21 @@ func (c *client) WaitForClusterInitialized(clusterName, namespace string) error
 		err = currentClusterInfo.RetrievalError
 
 		if err == nil {
-			// If cluster's ReadyCondition is False and severity is Error, it implies non-retriable error, so return error
+			// should retry in this condition
+			// For dimension env, sometimes it will take some time to connect to vcenter to create resources like cluster scret
 			if conditions.IsFalse(currentClusterInfo.ClusterObject, capi.ReadyCondition) &&
 				(*conditions.GetSeverity(currentClusterInfo.ClusterObject, capi.ReadyCondition) == capi.ConditionSeverityError) {
-				return true, errors.Errorf("cluster creation failed, reason:'%s', message:'%s'",
-					conditions.GetReason(currentClusterInfo.ClusterObject, capi.ReadyCondition),
-					conditions.GetMessage(currentClusterInfo.ClusterObject, capi.ReadyCondition))
+				reason := conditions.GetReason(currentClusterInfo.ClusterObject, capi.ReadyCondition)
+				message := conditions.GetMessage(currentClusterInfo.ClusterObject, capi.ReadyCondition)
+
+				maxTimeoutCounter++
```

Comment:
In case there is a real error, will this change make the creation failure after 3*c.operationTimeout? How about retry 4 times which is 1 minute?

Reply:
I re-use the `maxTimeoutCounter` for other conditions in `WaitForClusterInitialized`. Seems it's a reasonable timeout val.

Reply:
Sorry, what I mean is exactly Jeff's comment below: "so the "side-effect" is: for a real "error" level failure, it takes more time to fail ?" How about retry 4 times which is 1 minute?